### PR TITLE
[CARBONDATA-1357] fix convert bug

### PR DESCRIPTION
**Pull request by lionelcao**
fix byte[] to string convert bug

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by ravipesala** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/754/

**Comment by ravipesala** [comment]
Can one of the admins verify this patch?

**Comment by ravipesala** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3352/

**Comment by lionelcao** [comment]
retest this please

**Comment by ravipesala** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3354/

**Comment by lionelcao** [comment]
No failures found.

File: core/src/main/java/org/apache/carbondata/core/util/DataTypeConverterImpl.java [hunk 1]
```diff
@@ -29,7 +31,7 @@ public Object convertToDecimal(Object data) {
   }
 
   public Object convertFromByteToUTF8String(Object data) {
-    return data.toString();
+    return new String((byte[]) data, CarbonCommonConstants.DEFAULT_CHARSET_CLASS);
```
Comment (zzcclp) [inline core/src/main/java/org/apache/carbondata/core/util/DataTypeConverterImpl.java]: There is a java style error, try to remove the blank before 'data'.

**Comment by zzcclp** [comment]
there are some wrong with apache github, all prs builded with Spark 2.1.0 are failed.

**Comment by ravipesala** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/756/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**